fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/automatic/PrefAutomaticTranslationOptions.kt

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations.preferences.automatic
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.RadioButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Firefox Automatic Translation Options fragment compose view.
+ *
+ * @param selectedOption Selected option that will come from the translations engine.
+ */
+@Composable
+fun AutomaticTranslationOptions(
+    selectedOption: PrefAutomaticTranslationOption,
+) {
+    val optionsList = arrayListOf(
+        PrefAutomaticTranslationOption.OfferToTranslate(),
+        PrefAutomaticTranslationOption.AlwaysTranslate(),
+        PrefAutomaticTranslationOption.NeverTranslate(),
+    )
+    val selected = remember { mutableStateOf(selectedOption) }
+    Column(
+        modifier = Modifier
+            .background(
+                color = FirefoxTheme.colors.layer1,
+            ),
+    ) {
+        LazyColumn {
+            items(optionsList) { item: PrefAutomaticTranslationOption ->
+                AutomaticTranslationOptionRadioButton(
+                    text = stringResource(item.titleId),
+                    subText = stringResource(
+                        item.summaryId.first(),
+                        stringResource(item.summaryId.last()),
+                    ),
+                    selected = selected.value == item,
+                    onClick = {
+                        selected.value = item
+                    },
+                )
+            }
+        }
+    }
+}
+
+/**
+ * Top level radio button for the automatic translation options.
+ *
+ * @param text The main text to be displayed.
+ * @param subText The subtext to be displayed.
+ * @param selected [Boolean] that indicates whether the radio button is currently selected.
+ * @param onClick Invoked when the radio button is clicked.
+ */
+@Composable
+fun AutomaticTranslationOptionRadioButton(
```

Comment:
I am gonna ask for a follow up bug here where we extend our existing [ListItem](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt#216) to add the Radio Button use case. We should try to avoid implementing one-off components when we know the source of these components are coming from the design system. In this case, this is an extension of our existing ListItem.

To visualize this, you can click on "Go to main component" when inspecting a component in figma. 
<img width="605" alt="Screen Shot 2023-11-27 at 11 31 14 AM" src="https://github.com/mozilla-mobile/firefox-android/assets/1190888/d10c2276-b3f4-4d8d-9105-f36ebc6fce9d">

This will take you to https://www.figma.com/file/ujdiReypSypu2GhRgzArea/Android-Components?node-id=3016%3A12658&mode=dev. We are trying to build out the design components composable that we are using.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1868105